flink-clients/src/main/java/org/apache/flink/client/StreamGraphTranslator.java

```diff
@@ -40,14 +40,20 @@ public class StreamGraphTranslator implements FlinkPipelineTranslator {
 
     private static final Logger LOG = LoggerFactory.getLogger(StreamGraphTranslator.class);
 
+    private final ClassLoader userClassloader;
+
+    public StreamGraphTranslator(ClassLoader userClassloader) {
+        this.userClassloader = userClassloader;
+    }
+
     @Override
     public JobGraph translateToJobGraph(
```

Comment:
Looks to me it's better to pass in `userClassloader` as a parameter of this method, instead of making it a field of `StreamGraphTranslator.java`. In this way, we no longer need to change `FlinkPipelineTranslationUtil#translateToJSONExecutionPlan()` and its usages.